components/automate-ui/src/app/modules/infra-proxy/infra-environment-constraint/infra-environment-constraint.component.ts

```diff
@@ -0,0 +1,188 @@
+import {
+  Component,
+  OnInit,
+  ViewChild,
+  Input,
+  Output,
+  EventEmitter
+} from '@angular/core';
+
+import { FormBuilder, Validators, FormGroup } from '@angular/forms';
+import { Regex } from 'app/helpers/auth/regex';
+
+import { MatSelect } from '@angular/material/select';
+import { DynamicGrid } from '../create-environment-modal/create-environment-modal.component';
+
+@Component({
+  selector: 'app-infra-environment-constraint',
+  templateUrl: './infra-environment-constraint.component.html',
+  styleUrls: ['./infra-environment-constraint.component.scss']
+})
+
+export class InfraEnvironmentConstraintComponent implements OnInit {
+
+  @Input() constraintKeys: string[] = [];
+  @Input() dynamicArray: Array<DynamicGrid> = [];
+  @Input() name_id: string;
+  @Output() constraintValues: EventEmitter<DynamicGrid[]> = new EventEmitter();
+  @ViewChild(MatSelect) select: MatSelect;
+
+  public conflictError = false;
+  public operator_id = '';
+  public operatorKeys: string[] = [];
+  public secondFormGroup: FormGroup;
+  public selectedLangs: string[] = [];
+  public rowKeys: string[] = [];
+
+  constructor(
+    private fb: FormBuilder
+  ) {
+    this.secondFormGroup = this.fb.group({
+      version: ['', [Validators.required, Validators.pattern(Regex.patterns.VALID_VERSION)]]
+    });
+
+    this.operatorKeys = ['~>', '>=', '>', '=', '<', '<='];
+
+    this.operator_id = this.operatorKeys[0];
+  }
+
+  ngOnInit() {
+
+    this.conflictError = false;
+    this.constraintKeys.forEach(element =>
+
+      this.rowKeys.push(element)
+    );
+  }
+
+  selected(value, i) {
+
+    let previousValue;
+
+    this.selectedLangs.forEach((element, index) => {
+      if (index === i) {
+        previousValue = element;
+        if (value && value !== 'undefined') {
+          this.selectedLangs[i] = value;
+        }
+      }
+    });
+
+    this.dynamicArray.forEach((_, index) => {
+      if (index === i) {
+        this.dynamicArray[i].name = value;
+      }
+
+    });
+
+    this.constraintKeys.forEach((element, index) => {
+      if (element === value) {
+        this.constraintKeys.splice(index, 1);
+      }
+    });
+
+    this.constraintKeys.push(previousValue);
+    this.constraintKeys.sort((a, b) => a < b ? -1 : a > b ? 1 : 0);
+    this.name_id = this.constraintKeys[0];
+    this.constraintValues.emit(this.dynamicArray);
+  }
+
+  isSelected(lang: string) {
+    return this.selectedLangs.includes(lang);
+  }
+
+  public handleInput(event: KeyboardEvent): void {
+    if (this.isNavigationKey(event)) {
+      return;
+    }
+    this.conflictError = false;
+  }
+
+  // Handle the version change
+  public handleVersion(event, i) {
```

Comment:
Delete the comment; it does not add any information.
```suggestion
  public handleVersionChange(event, i) {
```

Same thing for `operator` and `name` below.

Reply:
added the changes